### PR TITLE
v0.2 for orka-deploy task

### PR DESCRIPTION
**Pull request by jkomoda**
# Changes

Added version `0.2` for `orka-deploy` task

* Compatibility changes to make the pipelines also work properly when using Tekton v0.34.1
* Added additional task parameters for feature parity with Orka v2.1.0:
  * `system-serial`: Assign an owned macOS system serial number to the VM (on Intel nodes only).
  * `vm-metadata`: Inject custom metadata to the VM (on Intel nodes only)
  * `gpu-passthrough`: Enables or disables GPU passthrough for the VM (on Intel nodes only).
* Changes in scripts to dynamically add additional parameters to VM config/deploy request when they are passed in pipelines.

# Notes
* Requires `task/orka-init/0.2/orka-init.yaml` in `pre-apply-task-hook.sh` in order for tests to work.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Follows the [authoring recommendations](https://github.com/tektoncd/catalog/blob/main/recommendations.md)
- [x] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality of task changed or new task added)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [x] Complies with [Catalog Organization TEP][TEP], see [example]. **Note** [An issue has been filed to automate this validation][validation]
  - [x] File path follows  `<kind>/<name>/<version>/name.yaml`
  - [x] Has `README.md` at `<kind>/<name>/<version>/README.md`
  - [x] Has mandatory `metadata.labels` - `app.kubernetes.io/version` the same as the `<version>` of the resource
  - [x] Has mandatory `metadata.annotations` `tekton.dev/pipelines.minVersion`
  - [x] mandatory `spec.description` follows the convention

          ```

          spec:
            description: >-
              one line summary of the resource

              Paragraph(s) to describe the resource.
          ```

_See [the contribution guide](https://github.com/tektoncd/catalog/blob/master/CONTRIBUTING.md)
for more details._

---

[TEP]: https://github.com/tektoncd/community/blob/master/teps/0003-tekton-catalog-organization.md
[example]: https://github.com/tektoncd/catalog/tree/master/task/git-clone/0.1
[validation]:  https://github.com/tektoncd/catalog/issues/413

### CODE REVIEW

File: task/orka-deploy/0.2/orka-deploy.yaml [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+  name: orka-deploy
+  labels:
+    app.kubernetes.io/version: "0.2"
+  annotations:
+    tekton.dev/categories: Deployment
+    tekton.dev/pipelines.minVersion: "0.16.0"
+    tekton.dev/tags: "orka, macstadium, deploy, build"
+    tekton.dev/displayName: "MacStadium Orka Deploy"
+    tekton.dev/platforms: "linux/amd64"
+    tekton.dev/displayName: "orka deploy"
```
Comment (spikeburton) [inline task/orka-deploy/0.2/orka-deploy.yaml:14]: Looks like this annotation is here twice 🙂 why change the display name from `MacStadium Orka Deploy`?

File: task/orka-deploy/0.2/samples/use-ssh-key.yaml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+# kubectl create secret generic orka-ssh-key --from-file=id_rsa=/path/to/id_rsa --from-literal=username=<username>
+# ###
+---
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  name: use-ssh-key
+spec:
+  taskRef:
+    name: orka-full
```
Comment (spikeburton) [inline task/orka-deploy/0.2/samples/use-ssh-key.yaml:20]: Unless there is a reason not to, I think we should keep this example as a `PipelineRun`. The `orka-deploy` task is not used here at all, and a big part of the purpose of the example was to illustrate how to use the init / deploy / teardown tasks together

File: task/orka-deploy/0.2/orka-deploy.yaml (hunk 1 above)
Comment (jkomoda) [reply task/orka-deploy/0.2/orka-deploy.yaml:14]: keeping it consistent with other task displayNames, but didn't remove previous :(

File: task/orka-deploy/0.2/orka-deploy.yaml [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+      default: "false"
+    - name: delete-vm
+      type: string
+      description: |
+        Specifies whether to delete the VM after use when run in a pipeline.
+        You can discard build agents that are no longer needed to free up resources.
+
+        Set to false if you intend to clean up VMs after use manually.
+      default: "true"
+    - name: docker-image
```
Comment (vinamra28) [inline task/orka-deploy/0.2/orka-deploy.yaml:50]: `orka-image` seems to be a better choice as compared to `docker-image`

File: task/orka-deploy/0.2/tests/resources.yaml [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: orka-svc
+  namespace: orka-deploy-0-2
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: orka-runner-deploy
```
Comment (vinamra28) [inline task/orka-deploy/0.2/tests/resources.yaml:47]: please refer to the comment https://github.com/tektoncd/catalog/pull/1030/files#r939492334

**Review by vinamra28 — APPROVED** [review]
/lgtm

**Pull request merged**